share/tasks.py

```diff
@@ -1,409 +1,187 @@
-
-        logger.info('Successfully submitted change for %s', raw)
-
-    def log_values(self):
-        return {
-            **super().log_values(),
-            'app_label': self.config.label,
-            'app_version': self.config.transformer.version,
-        }
+            }
+        }, headers={'Authorization': raw.suid.source_config.source.user.authorization(), 'Content-Type': 'application/vnd.api+json'})
+    except Exception as e:
+        logger.exception('Failed to transform %r', raw)
+        raise self.retry(countdown=10, exc=e)
+
+    if (resp.status_code // 100) != 2:
+        raise self.retry(countdown=10, exc=Exception('Unable to submit change graph. Received {!r}, {}'.format(resp, resp.content)))
+
+    logger.info('Successfully submitted change for %s', raw)
+
+
+@celery.shared_task(bind=True, retries=5)
+def disambiguate(self, normalized_id):
+    normalized = NormalizedData.objects.get(pk=normalized_id)
+    # Load all relevant ContentTypes in a single query
+    ContentType.objects.get_for_models(*apps.get_models('share'), for_concrete_models=False)
+
+    with transaction.atomic():
+        cg = ChangeGraph(normalized.data['@graph'], namespace=normalized.source.username)
+        cg.process()
+        cs = ChangeSet.objects.from_graph(cg, normalized.id)
+        if cs and (normalized.source.is_robot or normalized.source.is_trusted or Source.objects.filter(user=normalized.source).exists()):
+            # TODO: verify change set is not overwriting user created object
+            cs.accept()
+
+
+@celery.shared_task(bind=True, retries=5)
+def harvest(self, log_id=None, ignore_disabled=False, ingest=True, exhaust=True, superfluous=False, force=False, limit=None):
+    """Complete the harvest of the given HarvestLog or next the next available HarvestLog.
+
+    Args:
+        log_id (int, optional): Harvest the given log. Defaults to None.
+            If the given log cannot be locked, the task will retry indefinitely.
+            If the given log belongs to a disabled or deleted Source or SourceConfig, the task will fail.
+        ingest (bool, optional): Whether or not to start the full ingest process for harvested data. Defaults to True.
+        exhaust (bool, optional): Whether or not to start another harvest task if one is found. Defaults to True.
+            Used to prevent a backlog of harvests. If we have a valid job, spin off another task to eat through
+            the rest of the queue.
+        superfluous (bool, optional): Re-ingest Rawdata that we've already collected. Defaults to False.
+
+    """
+    qs = HarvestLog.objects.all()
+
+    if log_id is not None:
+        logger.debug('Loading harvest log %d', log_id)
+        qs = qs.filter(id=log_id)
+    else:
+        logger.debug('log_id was not specified, searching for an available log.')
+
+        if not ignore_disabled:
+            qs = qs.exclude(
+                source_config__disabled=True,
```

Comment:
These should be put in two separate `exclude`s -- this excludes logs with a disabled config AND a deleted source, should be OR.